src/lumigo_tracer/wrappers/sql/sqlalchemy_wrapper.py

```diff
@@ -0,0 +1,81 @@
+import importlib
+import uuid
+
+from lumigo_tracer.libs.wrapt import wrap_function_wrapper
+from lumigo_tracer.lumigo_utils import (
+    lumigo_safe_execute,
+    get_logger,
+    lumigo_dumps,
+    get_current_ms_time,
+)
+from lumigo_tracer.spans_container import SpansContainer
+
+try:
+    from sqlalchemy.event import listen
+except Exception:
+    listen = None
+
+
+SQL_SPAN = "mySql"
+
+
+def _before_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    SpansContainer.get_span().add_span(
+        {
+            "id": str(uuid.uuid4()),
+            "type": SQL_SPAN,
+            "started": get_current_ms_time(),
+            "connectionParameters": {
+                "host": conn.engine.url.host or conn.engine.url.database,
+                "port": conn.engine.url.port,
+                "database": conn.engine.url.database,
+                "user": conn.engine.url.username,
+            },
+            "query": lumigo_dumps(statement),
+            "values": lumigo_dumps(parameters),
+        }
+    )
+
+
+def _after_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    span = SpansContainer.get_span().get_last_span()
+    if not span:
+        get_logger().warning("Redis span ended without a record on its start")
+        return
+    span.update({"ended": get_current_ms_time(), "response": ""})
+
+
+def _handle_error(context):
+    span = SpansContainer.get_span().get_last_span()
+    if not span:
+        get_logger().warning("Redis span ended without a record on its start")
+        return
+    span.update(
+        {
+            "ended": get_current_ms_time(),
+            "error": lumigo_dumps(
+                {
+                    "type": context.original_exception.__class__.__name__,
+                    "args": context.original_exception.args,
```

Comment:
Are "name", and "args" enough? Don't we have more data about the error?

Reply:
unfortunately no,
This is the object that they give us (this is not an exception object!)